src/code42cli/cmds/cases.py

```diff
@@ -0,0 +1,246 @@
+    """For managing cases and events associated with cases."""
+    pass
+
+
+@cases.command()
+@click.argument("name", type=str)
+@assignee_option
+@description_option
+@notes_option
+@subject_option
+@sdk_options()
+def create(state, name, subject, assignee, description, notes):
+    """Create a new case."""
+    response = state.sdk.cases.create(
+        name,
+        subject=subject,
+        assignee=assignee,
+        description=description,
+        findings=notes,
+    )
+    return response
+
+
+@cases.command()
+@case_number_arg
+@name_option
+@assignee_option
+@description_option
+@notes_option
+@subject_option
+@status_option
+@sdk_options()
+def update(state, case_number, name, subject, assignee, description, notes, status):
+    """Update case details for the given case."""
+    state.sdk.cases.update(
+        case_number,
+        name=name,
+        subject=subject,
+        assignee=assignee,
+        description=description,
+        findings=notes,
+        status=status,
+    )
+
+
+def _parse_cases(pages):
+    cases = [case for page in pages for case in page["cases"]]
+    if not cases:
+        click.echo("No cases found.")
+    return cases
+
+
+@cases.command("list")
+@click.option(
+    "--name",
+    required=False,
+    help="Filter by name of a case, supports partial name matches.",
+)
+@click.option("--assignee", required=False, help="Filter by user UID of assignee.")
+@click.option(
+    "--subject", required=False, help="Filter by user UID of the subject of a case."
+)
+@click.option("--assignee", required=False, help="Filter by user UID of assignee.")
+@click.option(
+    "--min-create-time",
+    required=False,
+    help="Fetch cases created after given date time.",
+)
+@click.option(
+    "--max-create-time",
+    required=False,
+    help="Fetch cases created before given date time.",
+)
+@click.option(
+    "--min-update-time",
+    required=False,
+    help="Fetch cases last updated after given date time.",
+)
+@click.option(
+    "--max-update-time",
+    required=False,
+    help="Fetch cases last updated before given date time.",
+)
+@click.option("--status", required=False, help="Filter cases by case status.")
+@format_option
+@sdk_options()
+def _list(
+    state,
+    name,
+    assignee,
+    subject,
+    min_create_time,
+    max_create_time,
+    min_update_time,
+    max_update_time,
+    status,
+    format,
+):
+    """List all the cases."""
+    pages = state.sdk.cases.get_all(
+        name=name,
+        assignee=assignee,
+        subject=subject,
+        min_create_time=min_create_time,
+        max_create_time=max_create_time,
+        min_update_time=min_update_time,
+        max_update_time=max_update_time,
+        status=status,
+    )
+    formatter = OutputFormatter(format, _HEADER_KEYS_MAP)
+    cases = _parse_cases(pages)
+    if cases:
+        formatter.echo_formatted_list(cases)
+
+
+@cases.command()
+@case_number_arg
+@sdk_options()
+def show(state, case_number):
```

Comment:
`show` commands are supposed to be more table like, not just echoing JSON.
See other show commands in the CLI as an example